src/app.ts

```diff
@@ -143,6 +185,28 @@ class FrameworkClient {
 			return command;
 		});
 		const imported = await Promise.all(commandsImports);
+
+		// Check to see if we are forced to reset all slash commands.
+		// This can be useful in the event a bad command got registered and needs to be removed.
+		if (this.options.slashCommandReset) {
+			await this.deleteSlashCommands();
+		}
+
+		// Loop over all command and verify if one of them is a slash command.
+		// If a command is one, it is moved to the slashCommands array and removed from the imported array.
+		// Then, it will register itself as a slash command.
+		imported.forEach(async element => {
+			if(element instanceof Command) {
+				if(element.slashCommand) {
+					await this.loadSlashCommand(element);
+
+					// Yes, I am doing this on this and yes, this is not a good idea. It just gotta work now.
+					this.slashCommands.push(element);
+					imported.splice(imported.indexOf(element), 1);
```

Comment:
If there are two slash commands back to back, this will only load the first one then skip the second one, don't modify loops while reading them